src/main/java/toughasnails/config/GameplayConfigurationHandler.java

```diff
@@ -8,26 +8,131 @@
+            			new BlockTemperatureData("minecraft:lava", 1.5F, new BlockTemperatureStateData[0]),
+            			new BlockTemperatureData("minecraft:flowing_lava", 1.5F, new BlockTemperatureStateData[0]),
+            		};
+            		
+            		String defaultTempJsonString = serializer.toJson(defaultTemperatureDataArray);
+            		
+            		FileUtils.writeStringToFile(temperatureConfigFile, defaultTempJsonString);
+            	}
+        		
+            	String jsonString = FileUtils.readFileToString(temperatureConfigFile);
+            	
+        		try
+        		{
+        			Gson gson = new Gson();
+        			
+        			BlockTemperatureData[] tempDataArray = gson.fromJson(jsonString, BlockTemperatureData[].class);
+        			
+        			for (BlockTemperatureData tempDataObj : tempDataArray)
+        			{
+        				//TODO: Clever merging, or just rely on user to not duplicate?
+        				blockTemperatureData.put(tempDataObj.blockName, tempDataObj);
+        			}
+        			
+        		}
+        		catch (Exception e)
+        		{
+        			ToughAsNails.logger.error("Error parsing block temperature config from json: " + blockConfigName, e);
+        		}
+				
+			}
+            catch (Exception e)
+            {
+            	ToughAsNails.logger.error("Error parsing block temperature config: " + blockConfigName, e);
+			}
+            
+            
+            //Material temperature config:
+            materialTemperatureData = new MaterialTemperatureData();
+            
+            String materialConfigName = FilenameUtils.getFullPath(configFile.getPath()) + "materialTemperatureConfig.json";
+            
+            try
+            {
+            	File temperatureConfigFile = new File(materialConfigName);
+            	
+            	if (!temperatureConfigFile.exists())
+            	{
+            		Gson serializer = new GsonBuilder().setPrettyPrinting().create();
+            		
+            		String defaultTempJsonString = serializer.toJson(materialTemperatureData);
+            		
+            		FileUtils.writeStringToFile(temperatureConfigFile, defaultTempJsonString);
+            	}
+            	
+            	String jsonString = FileUtils.readFileToString(temperatureConfigFile);
+            	
+            	try
+            	{
+            		Gson gson = new Gson();
+            		
+            		materialTemperatureData = gson.fromJson(jsonString, MaterialTemperatureData.class);
+            	}
+            	catch (Exception e)
+            	{
+            		ToughAsNails.logger.error("Error parsing material temperature config from json: " + materialConfigName, e);
+            	}
+            	
+            }
+            catch (Exception e)
+            {
+            	ToughAsNails.logger.error("Error parsing material temperature config: " + materialConfigName, e);
+			}
         }
     }
 
```

Comment:
Not just in this class, but other classes too, there's a lot of mixing and matching spaces and tabs. With TAN we only use spaces for all of our classes. There's an option in eclipse and Intelli J for this. This messes up rendering on Github in some browsers/operating systems.